datafusion/physical-expr/src/physical_expr.rs

```diff
@@ -128,6 +137,232 @@ impl PhysicalExprStats for BasicExpressionStats {
     }
 }
 
+/// a Trait for marking tree node types that are rewritable
+pub trait TreeNodeRewritable: Clone {
+    /// Transform the tree node using the given [TreeNodeRewriter]
+    /// It performs a depth first walk of an node and its children.
+    ///
+    /// For an node tree such as
+    /// ```text
+    /// ParentNode
+    ///    left: ChildNode1
+    ///    right: ChildNode2
+    /// ```
+    ///
+    /// The nodes are visited using the following order
+    /// ```text
+    /// pre_visit(ParentNode)
```

Comment:
I think these examples need to be change to use the terms `transform_down` and `transform_up` and `transform`

I also think it would be nice to use the same terms here as are used in the rewriter (previsit / postvisit) but I realize they don't quite match up.

Reply:
Can I keep the current naming ? As you had mentioned, they do not match.

Reply:
I think we need to make sure the doc comments and code match -- the naming of the code is fine with me (for the reasons you mention)